Userland/Libraries/LibGfx/TIFFGenerator.py

```diff
@@ -117,11 +117,11 @@ class ExtraSample(EnumWithExportName):
     Tag('277', [TIFFType.UnsignedShort], [1], None, "SamplesPerPixel", is_required=True),
     Tag('278', [TIFFType.UnsignedShort, TIFFType.UnsignedLong], [1], None, "RowsPerStrip", is_required=True),
     Tag('279', [TIFFType.UnsignedShort, TIFFType.UnsignedLong], [], None, "StripByteCounts", is_required=True),
-    Tag('282', [TIFFType.UnsignedRational], [], None, "XResolution"),
-    Tag('283', [TIFFType.UnsignedRational], [], None, "YResolution"),
-    Tag('284', [TIFFType.UnsignedShort], [], PlanarConfiguration.Chunky, "PlanarConfiguration", PlanarConfiguration),
+    Tag('282', [TIFFType.UnsignedRational], [1], None, "XResolution"),
```

Comment:
adobe layered tiffs might support more than one? but it's a nonstandard extension (…and i don't know how it's implemented – do we have something to dump tiff tags? if so i could check)

Reply:
Is that files with multiple pages? I think that's implemented with multiple IFD (so one XResolution tag in each IFD).
Just decode with `image` with `TIFF_DEBUG`

Reply:
```
 % Build/lagom/bin/image ~/Documents/layers.tif --no-output
Setting image file directory pointer to 8
Failed to decode input file '/Users/thakis/Documents/layers.tif'
```

[layers.tif.zip](https://github.com/SerenityOS/serenity/files/13796909/layers.tif.zip)

(both layers are just white – and don't worry too much about it, it's nonstandard. It's a two-layer image saved as tiff in photoshop)

Reply:
Looks like someone didn't compile with `TIFF_DEBUG`

```
Setting image file directory pointer to 8
UnknownTag(254, UnsignedLong): 0
ImageWidth(UnsignedShort): 400
ImageHeight(UnsignedShort): 300
BitsPerSample(UnsignedShort): [8, 8, 8, 8]
Compression(UnsignedShort): NoCompression
PhotometricInterpretation(UnsignedShort): CMYK
StripOffsets(UnsignedLong): 73112
Orientation(UnsignedShort): Default
SamplesPerPixel(UnsignedShort): 4
RowsPerStrip(UnsignedShort): 300
StripByteCounts(UnsignedLong): 480000
XResolution(UnsignedRational): 300 (3000000/10000)
YResolution(UnsignedRational): 300 (3000000/10000)
PlanarConfiguration(UnsignedShort): Chunky
ResolutionUnit(UnsignedShort): Inch
UnknownTag(305, ASCII): Adobe Photoshop 25.1 (Macintosh)
UnknownTag(306, ASCII): 2023:12:29 21:38:32
```

The decoder fails because the next tag (19/23) is this one, and we error because of an unknown type (IFD):
https://www.awaresystems.be/imaging/tiff/tifftags/subifds.html
![image](https://github.com/SerenityOS/serenity/assets/26030965/410559d4-9675-4dee-90a5-a8275c85effb)

I'm pretty sure this is required by the spec to skip any unknown tag. And as people like to create non-standard TIFF, we should be more robust regarding this. But to answer the initial question: no, even layered tiffs have a single value inside `XResolution`.

Reply:
I _did_ add `#define TIFF_DEBUG 1` to TIIFLoader.cpp. Is that the wrong file?

Reply:
You need to add it to both TIFFTagHandler.cpp and TIFFLoader.cpp to get the full debug